### DISCUSSION TITLE
[question] wanna ask tauri app can run on browser or not?

### DISCUSSION THREAD

**Discussion by lucasjinreal** [tauri-apps/tauri#6010] [Ideas]
Since it was using js, does it able to serve on browser or not? (not just desktop), if positive, is it hard to do so?

**Comment by CodeDead** [comment]
You can make your app cross-compatible very easily. I have an example project that runs in both Tauri and the browser here

https://github.com/CodeDead/Advanced-PassGen/

**Comment by AndrewJSchoen** [comment]
@CodeDead are you creating a web worker that executes the same tauri-based functions, or are you re-implementing the calls elsewhere, and wrapping those in the worker?

**Reply by CodeDead** [reply]
@AndrewJSchoen If it's running on the web, it will perform the functions in a webworker, if it's running on a desktop environment, it will run the rust native code.

The rust code will always be faster than the JS webworkers, but it's the most elegant solution in my opinion.

Example:
https://github.com/CodeDead/Advanced-PassGen/blob/master/src/reducers/MainReducer/Actions/index.js#L48

**Reply by AndrewJSchoen** [reply]
@CodeDead Makes sense, I guess I was wondering specifically about whether you had to duplicate your code, or if you had a mechanism for generating a wasm-built library from the tauri hooks in src-tauri. Thanks for the quick reply!

**Reply by CodeDead** [reply]
> @CodeDead Makes sense, I guess I was wondering specifically about whether you had to duplicate your code, or if you had a mechanism for generating a wasm-built library from the tauri hooks in src-tauri. Thanks for the quick reply!

Good idea but keep in mind that WASM doesn't have all the appropriate capabilities to access, for example, the filesystem or anything outside the browser environment. There would have to be one reason or another to make a desktop application that requires you to access data, native API's or something like that in order to justify the app being a desktop application (unless if you include a runtime that allows wasm code access outside of the sandbox)

**Reply by AndrewJSchoen** [reply]
Speaking only for our use, we actually are currently porting a system from web _to_ tauri, mainly in the interest of increasing performance and making things more robust. Currently, we have some rust-based wasm code (which needs to be rust because of performance), which is utilized within a web-worker. Ideally, we would like to move over some of that functionality to the tauri backend (which is currently not utilized), but not lose the ability to run in the browser, since that is still helpful in certain circumstances.

**Reply by XRTIDE** [reply]
@AndrewJSchoen This is an example that uses WASM in a web-worker thread, and be advised:
https://www.sitepen.com/blog/using-webassembly-with-web-workers
